### PR TITLE
deps: bump `org_llvm_releases_compiler_rt` -> 15.0.01

### LINKED ISSUE #23179: Newer release available `org_llvm_releases_compiler_rt`: llvmorg-15.0.1 (current: llvmorg-14.0.0)
Package Name: org_llvm_releases_compiler_rt@14.0.0
Current Version: llvmorg-14.0.0@2022-03-23
Available Version: llvmorg-15.0.1@2022-09-20
Upstream releases: https://github.com/llvm/llvm-project/releases

### PR DESCRIPTION
**Pull request by RiverPhillips**
Commit Message: deps: bump `org_llvm_releases_compiler_rt` -> 15.0.01
Additional Description: Resolves #23179
Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by keith — APPROVED** [review]

**Comment by phlax** [comment]
@keith @lizan im wondering if this needs to match what is in the build image

**Comment by RiverPhillips** [comment]
@phlax I believe they're already different

**Comment by keith** [comment]
they are actually the same right now, i think we do need to match them, i forgot that when i reviewed initially

**Comment by RiverPhillips** [comment]
Am I correct in assuming you meant to resolve #23178 as part of this too?

That's the only other LLVM issue I can see referencing v15.0.1

**Comment by keith** [comment]
Ah there are a lot of versions here. this is the one I was talking about https://github.com/envoyproxy/envoy-build-tools/blob/0f17a22c7902e21e5e003f5296adb628d2f16b2f/build_container/build_container_ubuntu.sh#L84

**Comment by RiverPhillips** [comment]
Cheers, sorry about the confusion. I've [opened a PR](https://github.com/envoyproxy/envoy-build-tools/pull/185) to resolve that

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -1191,12 +1191,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "compiler-rt",
         project_desc = "LLVM compiler runtime library",
         project_url = "https://compiler-rt.llvm.org",
-        version = "14.0.0",
-        sha256 = "27ab7fcfb21d108093c0be766a9ed5fe18c04e4f74f936069711a312c8ae0377",
+        version = "15.0.1",
```
Comment (yanavlasov) [inline bazel/repository_locations.bzl:1194]: Is this right? We still on clang 14.0.0? I think we should update these together with the clang.

**Review by yanavlasov — COMMENTED** [review]
/wait-any

**Comment by phlax** [comment]
this is waiting on the update to envoy-build-tools - but i think we dont want to push that through until after the 1.24 release

/wait

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (moderation) [reply bazel/repository_locations.bzl:1194]: `15.0.2` is out - https://github.com/llvm/llvm-project/releases/tag/llvmorg-15.0.2

**Pull request closed without merging**